flink-libraries/flink-state-processing-api/src/test/java/org/apache/flink/state/api/output/SnapshotUtilsTest.java

```diff
@@ -45,18 +49,51 @@
     private static final List<String> EXPECTED_CALL_OPERATOR_SNAPSHOT =
             Arrays.asList("prepareSnapshotPreBarrier", "snapshotState", "notifyCheckpointComplete");
 
+    private static final List<SnapshotType> EXPECTED_SNAPSHOT_TYPE = new ArrayList<>();
+
     @Rule public TemporaryFolder folder = new TemporaryFolder();
 
     private static final List<String> ACTUAL_ORDER_TRACKING =
             Collections.synchronizedList(new ArrayList<>(EXPECTED_CALL_OPERATOR_SNAPSHOT.size()));
 
+    private static final List<SnapshotType> ACTUAL_SNAPSHOT_TYPE = new ArrayList<>();
+
+    @Before
+    public void initializeVariable() {
+        ACTUAL_ORDER_TRACKING.clear();
+        EXPECTED_SNAPSHOT_TYPE.clear();
+        ACTUAL_SNAPSHOT_TYPE.clear();
+    }
+
     @Test
     public void testSnapshotUtilsLifecycle() throws Exception {
+        EXPECTED_SNAPSHOT_TYPE.add(SavepointType.savepoint(SavepointFormatType.DEFAULT));
         StreamOperator<Void> operator = new LifecycleOperator();
         Path path = new Path(folder.newFolder().getAbsolutePath());
 
         SnapshotUtils.snapshot(operator, 0, 0L, true, false, new Configuration(), path);
 
+        Assert.assertEquals(EXPECTED_SNAPSHOT_TYPE, ACTUAL_SNAPSHOT_TYPE);
+        Assert.assertEquals(EXPECTED_CALL_OPERATOR_SNAPSHOT, ACTUAL_ORDER_TRACKING);
+    }
+
+    @Test
+    public void testSnapshotUtilsLifecycleWithNativeSavepoint() throws Exception {
```

Comment:
I think we can reduce duplicate code to make one private method `testSnapshotUtilsLifecycle(SavepointFormatType savepointFormatType)` and let two public test mthod just call it via `testSnapshotUtilsLifecycle(SavepointFormatType.DEFAULT)` and `testSnapshotUtilsLifecycle(SavepointFormatType.NATIVE)`.

By doing so, we can also drop the `initializeVariable` method.